var/spack/repos/builtin/packages/grid/package.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+# ----------------------------------------------------------------------------
+# If you submit this package back to Spack as a pull request,
+# please first remove this boilerplate and all FIXME comments.
+#
+# This is a template package file for Spack.  We've put "FIXME"
+# next to all the things you'll want to change. Once you've handled
+# them, you can save this file and test your package like this:
+#
+#     spack install grid
+#
+# You can edit this file again by typing:
+#
+#     spack edit grid
+#
+# See the Spack documentation for more information on packaging.
+# ----------------------------------------------------------------------------
+
+from spack.package import *
+
+
+class Grid(AutotoolsPackage):
+    """Data parallel C++ mathematical object library."""
+
+    homepage = "https://github.com/paboyle/Grid"
+    url      = "https://github.com/paboyle/Grid/archive/refs/tags/0.8.2.tar.gz"
+    git      = "https://github.com/paboyle/Grid.git"
+
+    version('develop', branch='develop')
+    # v0.8.2 is currently the latest "numbered" version, but it doesn't build
+    # because during bootstrap it automatically downloads Eigen, but it fetches
+    # it from the old Bitbucket repository which doesn't exist anymore.
+    version('0.8.2', sha256='2b3389a75dec78f154aa28c8828d81a251b0331284f37cede95747255adef13a', deprecated=True)
+
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool',  type='build')
+    depends_on('m4',       type='build')
+
+    depends_on('gmp')
+    depends_on('mpfr')
+
+    variant('comms', default='mpi',
+            values=('none', 'mpi', 'mpi3', conditional('shmem', when='^cray-mpich')),
+            description='Choose communication interface')
+    depends_on('mpi', when='comms=mpi')
+    depends_on('cray-mpich', when='comms=shmem')
+    depends_on('mpi@3:', when='comms=mpi3')
+
+    variant('fftw', default=True, description='Activate FFTW support')
+    depends_on('fftw-api@3', when='+fftw')
+
+    variant('lapack', default=False, description='Activate LAPACK support')
+    depends_on('lapack', when='+lapack')
+
+    variant('hdf5', default=False, description='Activate HDF5 support')
+    depends_on('hdf5', when='+hdf5')
+
+    variant('lime', default=False, description='Activate LIME support')
+    depends_on('c-lime', when='+lime')
+
+    variant('doc', default=False,
+            description='Build the documentation with doxygen')
+    depends_on("doxygen", type="build", when="+doc")
+
+    variant('gen-simd-width', default='64',
+            description='Size (in bytes) of the generic SIMD vector type')
```

Comment:
Is it fine to accept anything here, or should we restrict the possible values?

Reply:
I presume nice powers of 2 would be the possible values, but I didn't find in https://spack.readthedocs.io/en/latest/packaging_guide.html#variants how to restrict numerical values, apart from listing all of them.  I searched a bit in Grid code and doesn't seem to me this value is validated anywhere, so can't tell what are the precise expected values.  Also, this is relevant only for the generic SIMD vector type, which is quite unlikely to be used (unless someone purposefully targets the generic architecture).